model-optimizer/mo/utils/cli_parser.py

```diff
@@ -793,6 +811,119 @@ def parse_input_value(input_value: str):
     return node_name, shape, value, data_type
 
 
+def split_str_avoiding_square_brackets(s: str) -> list:
+    """
+    Splits a string by comma, but skips commas inside square brackets.
+    :param s: string to split
+    :return: list of strings split by comma
+    """
+    res = list()
+    skipping = 0
+    last_idx = 0
+    for i, c in enumerate(s):
+        if c == '[':
+            skipping += 1
+        elif c == ']':
+            skipping -= 1
+        elif c == ',' and skipping == 0:
+            res.append(s[last_idx:i])
+            last_idx = i + 1
+    res.append(s[last_idx:])
+    return res
+
+
+def split_layouts_by_arrow(s: str) -> tuple:
+    """
+    Splits a layout string by first arrow (->).
+    :param s: string to split
+    :return: tuple containing source and target layouts
+    """
+    arrow = s.find('->')
+    if arrow != -1:
+        source_layout = s[:arrow]
+        target_layout = s[arrow + 2:]
+        if source_layout == '':
+            source_layout = None
+        if target_layout == '':
+            target_layout = None
+        return source_layout, target_layout
+    else:
+        return s, None
+
+
+def parse_layouts_by_destination(s: str, parsed: dict, dest: str = None) -> None:
+    """
+    Parses layout command line to get all names and layouts from it. Adds all found data in the 'parsed' dict.
+    :param s: string to parse
+    :param parsed: dict where result will be stored
+    :param dest: type of the command line:
+      * 'source' is --source_layout
+      * 'target' is --target_layout
+      * None is --layout
+    """
+    list_s = split_str_avoiding_square_brackets(s)
+    if len(list_s) == 1 and (list_s[0][-1] not in ')]' or (list_s[0][0] == '[' and list_s[0][-1] == ']')):
+        # single layout case
+        s_layout = None
+        t_layout = None
+        if '' in parsed:
+            s_layout = parsed['']['source_layout']
+            t_layout = parsed['']['target_layout']
+        if dest == 'source':
+            s_layout = list_s[0]
+        elif dest == 'target':
+            t_layout = list_s[0]
+        else:
+            s_layout, t_layout = split_layouts_by_arrow(list_s[0])
+        parsed[''] = {'source_layout': s_layout, 'target_layout': t_layout}
+    else:
+        for layout_str in list_s:
+            s_layout = None
+            t_layout = None
+            p1 = re.compile(r'(\S+)\((\S+)\)')
+            m1 = p1.match(layout_str)
+            p2 = re.compile(r'(\S+)(\[\S+\])')
+            m2 = p2.match(layout_str)
+            if m1:
+                found_g = m1.groups()
+            elif m2:
+                found_g = m2.groups()
+            else:
+                raise Error("More then one layout provided for --{}layout without providing name.".format(
+                    dest + '_' if dest else ''))
+            if found_g[0] in parsed:
```

Comment:
minor: here code do the same as in line 869, possibly create function and re-use

Reply:
I also thought about doing that, but it seems that by doing that I will make code harder to read